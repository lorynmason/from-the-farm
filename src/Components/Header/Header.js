import React from 'react';

export const Header = () => {
  return (
    <header>
      <h1>from the farm</h1>
      <img id="logo" src="https://www.shareicon.net/data/2016/06/18/596437_pig_512x512.png" alt="from the farm cute pig logo"/>
      {/* <img id="logo" src="../../../public/assets/596437_pig_512x512.png" alt="from the farm cute pig logo"/> */}
    </header>
  )
}