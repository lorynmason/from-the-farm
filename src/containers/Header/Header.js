import React from 'react';
import { Link } from 'react-router-dom';
import image from '../../styles/images/pig-logo.png'
import { removeUser, isLoading, addMessage } from '../../actions';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export const Header = (props) => {
  const clickBuy = () => {
    props.isLoading(true)
    setTimeout(() => {
      props.isLoading(false)
    }, 1000);
  }

  const signout = () => {
    props.removeUser();
    props.addMessage('You are now signed-out');
  }

  return (
    <header>
      <div className="header-logo">
      <Link to="/" onClick={clickBuy} title="Click to go to Home Page">
        <h1>from the farm</h1>
        <img id="logo" src={image} alt="from the farm cute pig logo"/>
      </Link>
      </div>
      <nav className="menu">
        <input type="checkbox" href="#" className="menu-open" name="menu-open" id="menu-open"/>
          <label className="menu-open-button" htmlFor="menu-open">
            <span className="hamburger hamburger-1"></span>
            <span className="hamburger hamburger-2"></span>
            <span className="hamburger hamburger-3"></span>
          </label>
          <Link to="/buy" onClick={clickBuy}><i className="fas fa-carrot"><p>buy</p></i></Link>
          <Link to="/profile"><i className="fas fa-user-circle"><p>my profile</p></i></Link>
          <Link to="/add-product"><i className="fas fa-plus-circle"><p>inventory</p></i></Link>
          <Link to="/" onClick={signout}><i className="fas fa-sign-out-alt"><p>sign out</p></i></Link>
      </nav>
    </header>
  );
}

export const mapDispatchToProps = (dispatch) => ({
  removeUser: () => dispatch(removeUser()),
  addMessage: (message) => dispatch(addMessage(message)),
  isLoading: (bool) => dispatch(isLoading(bool))
});

export default connect(null, mapDispatchToProps)(Header);

Header.propTypes = {
  removeUser: PropTypes.func,
  addMessage: PropTypes.func,
  isLoading: PropTypes.func
}
