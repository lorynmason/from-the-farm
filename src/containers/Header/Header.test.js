import React from 'react';
import { Header, mapDispatchToProps } from './Header';
import { shallow } from 'enzyme'

describe('header', () => {
  let mockFunc = jest.fn()
  it('should match snapshot', () => {
    let wrapper = shallow(<Header removeUser={mockFunc} addMessage={mockFunc} isLoading={mockFunc}/>)
    expect(wrapper).toMatchSnapshot()
  });

  describe('mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    it('should dispatch removeUser, addMessage, isLoading with the correct params when addProductsToStore is called', () => {
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.addMessage('You are now signed-out');
      mappedProps.removeUser();
      mappedProps.isLoading(true)

      expect(mockDispatch).toHaveBeenCalledTimes(3);
    });
  });
});
