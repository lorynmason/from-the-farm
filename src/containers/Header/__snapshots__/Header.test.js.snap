// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    addMessage={[MockFunction]}
    isLoading={[MockFunction]}
    removeUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header-logo"
        >
          <Link
            onClick={[Function]}
            replace={false}
            title="Click to go to Home Page"
            to="/"
          >
            <h1>
              from the farm
            </h1>
            <img
              alt="from the farm cute pig logo"
              id="logo"
              src="pig-logo.png"
            />
          </Link>
        </div>,
        <nav
          className="menu"
        >
          <input
            className="menu-open"
            href="#"
            id="menu-open"
            name="menu-open"
            type="checkbox"
          />
          <label
            className="menu-open-button"
            htmlFor="menu-open"
          >
            <span
              className="hamburger hamburger-1"
            />
            <span
              className="hamburger hamburger-2"
            />
            <span
              className="hamburger hamburger-3"
            />
          </label>
          <div
            className="nav-list"
          >
            <Link
              onClick={[Function]}
              replace={false}
              to="/buy"
            >
              <i
                className="fas fa-carrot"
              >
                <p>
                  buy
                </p>
              </i>
            </Link>
            <Link
              replace={false}
              to="/profile"
            >
              <i
                className="fas fa-user-circle"
              >
                <p>
                  my profile
                </p>
              </i>
            </Link>
            <Link
              replace={false}
              to="/add-product"
            >
              <i
                className="fas fa-plus-circle"
              >
                <p>
                  inventory
                </p>
              </i>
            </Link>
            <Link
              onClick={[Function]}
              replace={false}
              to="/"
            >
              <i
                className="fas fa-sign-out-alt"
              >
                <p>
                  sign out
                </p>
              </i>
            </Link>
          </div>
        </nav>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            onClick={[Function]}
            replace={false}
            title="Click to go to Home Page"
            to="/"
          >
            <h1>
              from the farm
            </h1>
            <img
              alt="from the farm cute pig logo"
              id="logo"
              src="pig-logo.png"
            />
          </Link>,
          "className": "header-logo",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h1>
                from the farm
              </h1>,
              <img
                alt="from the farm cute pig logo"
                id="logo"
                src="pig-logo.png"
              />,
            ],
            "onClick": [Function],
            "replace": false,
            "title": "Click to go to Home Page",
            "to": "/",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "from the farm",
              },
              "ref": null,
              "rendered": "from the farm",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "from the farm cute pig logo",
                "id": "logo",
                "src": "pig-logo.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="menu-open"
              href="#"
              id="menu-open"
              name="menu-open"
              type="checkbox"
            />,
            <label
              className="menu-open-button"
              htmlFor="menu-open"
            >
              <span
                className="hamburger hamburger-1"
              />
              <span
                className="hamburger hamburger-2"
              />
              <span
                className="hamburger hamburger-3"
              />
            </label>,
            <div
              className="nav-list"
            >
              <Link
                onClick={[Function]}
                replace={false}
                to="/buy"
              >
                <i
                  className="fas fa-carrot"
                >
                  <p>
                    buy
                  </p>
                </i>
              </Link>
              <Link
                replace={false}
                to="/profile"
              >
                <i
                  className="fas fa-user-circle"
                >
                  <p>
                    my profile
                  </p>
                </i>
              </Link>
              <Link
                replace={false}
                to="/add-product"
              >
                <i
                  className="fas fa-plus-circle"
                >
                  <p>
                    inventory
                  </p>
                </i>
              </Link>
              <Link
                onClick={[Function]}
                replace={false}
                to="/"
              >
                <i
                  className="fas fa-sign-out-alt"
                >
                  <p>
                    sign out
                  </p>
                </i>
              </Link>
            </div>,
          ],
          "className": "menu",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "menu-open",
              "href": "#",
              "id": "menu-open",
              "name": "menu-open",
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="hamburger hamburger-1"
                />,
                <span
                  className="hamburger hamburger-2"
                />,
                <span
                  className="hamburger hamburger-3"
                />,
              ],
              "className": "menu-open-button",
              "htmlFor": "menu-open",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "hamburger hamburger-1",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "hamburger hamburger-2",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "hamburger hamburger-3",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  onClick={[Function]}
                  replace={false}
                  to="/buy"
                >
                  <i
                    className="fas fa-carrot"
                  >
                    <p>
                      buy
                    </p>
                  </i>
                </Link>,
                <Link
                  replace={false}
                  to="/profile"
                >
                  <i
                    className="fas fa-user-circle"
                  >
                    <p>
                      my profile
                    </p>
                  </i>
                </Link>,
                <Link
                  replace={false}
                  to="/add-product"
                >
                  <i
                    className="fas fa-plus-circle"
                  >
                    <p>
                      inventory
                    </p>
                  </i>
                </Link>,
                <Link
                  onClick={[Function]}
                  replace={false}
                  to="/"
                >
                  <i
                    className="fas fa-sign-out-alt"
                  >
                    <p>
                      sign out
                    </p>
                  </i>
                </Link>,
              ],
              "className": "nav-list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <i
                    className="fas fa-carrot"
                  >
                    <p>
                      buy
                    </p>
                  </i>,
                  "onClick": [Function],
                  "replace": false,
                  "to": "/buy",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <p>
                      buy
                    </p>,
                    "className": "fas fa-carrot",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "buy",
                    },
                    "ref": null,
                    "rendered": "buy",
                    "type": "p",
                  },
                  "type": "i",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <i
                    className="fas fa-user-circle"
                  >
                    <p>
                      my profile
                    </p>
                  </i>,
                  "replace": false,
                  "to": "/profile",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <p>
                      my profile
                    </p>,
                    "className": "fas fa-user-circle",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "my profile",
                    },
                    "ref": null,
                    "rendered": "my profile",
                    "type": "p",
                  },
                  "type": "i",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <i
                    className="fas fa-plus-circle"
                  >
                    <p>
                      inventory
                    </p>
                  </i>,
                  "replace": false,
                  "to": "/add-product",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <p>
                      inventory
                    </p>,
                    "className": "fas fa-plus-circle",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "inventory",
                    },
                    "ref": null,
                    "rendered": "inventory",
                    "type": "p",
                  },
                  "type": "i",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <i
                    className="fas fa-sign-out-alt"
                  >
                    <p>
                      sign out
                    </p>
                  </i>,
                  "onClick": [Function],
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <p>
                      sign out
                    </p>,
                    "className": "fas fa-sign-out-alt",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "sign out",
                    },
                    "ref": null,
                    "rendered": "sign out",
                    "type": "p",
                  },
                  "type": "i",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "nav",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header-logo"
          >
            <Link
              onClick={[Function]}
              replace={false}
              title="Click to go to Home Page"
              to="/"
            >
              <h1>
                from the farm
              </h1>
              <img
                alt="from the farm cute pig logo"
                id="logo"
                src="pig-logo.png"
              />
            </Link>
          </div>,
          <nav
            className="menu"
          >
            <input
              className="menu-open"
              href="#"
              id="menu-open"
              name="menu-open"
              type="checkbox"
            />
            <label
              className="menu-open-button"
              htmlFor="menu-open"
            >
              <span
                className="hamburger hamburger-1"
              />
              <span
                className="hamburger hamburger-2"
              />
              <span
                className="hamburger hamburger-3"
              />
            </label>
            <div
              className="nav-list"
            >
              <Link
                onClick={[Function]}
                replace={false}
                to="/buy"
              >
                <i
                  className="fas fa-carrot"
                >
                  <p>
                    buy
                  </p>
                </i>
              </Link>
              <Link
                replace={false}
                to="/profile"
              >
                <i
                  className="fas fa-user-circle"
                >
                  <p>
                    my profile
                  </p>
                </i>
              </Link>
              <Link
                replace={false}
                to="/add-product"
              >
                <i
                  className="fas fa-plus-circle"
                >
                  <p>
                    inventory
                  </p>
                </i>
              </Link>
              <Link
                onClick={[Function]}
                replace={false}
                to="/"
              >
                <i
                  className="fas fa-sign-out-alt"
                >
                  <p>
                    sign out
                  </p>
                </i>
              </Link>
            </div>
          </nav>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              onClick={[Function]}
              replace={false}
              title="Click to go to Home Page"
              to="/"
            >
              <h1>
                from the farm
              </h1>
              <img
                alt="from the farm cute pig logo"
                id="logo"
                src="pig-logo.png"
              />
            </Link>,
            "className": "header-logo",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <h1>
                  from the farm
                </h1>,
                <img
                  alt="from the farm cute pig logo"
                  id="logo"
                  src="pig-logo.png"
                />,
              ],
              "onClick": [Function],
              "replace": false,
              "title": "Click to go to Home Page",
              "to": "/",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "from the farm",
                },
                "ref": null,
                "rendered": "from the farm",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "from the farm cute pig logo",
                  "id": "logo",
                  "src": "pig-logo.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="menu-open"
                href="#"
                id="menu-open"
                name="menu-open"
                type="checkbox"
              />,
              <label
                className="menu-open-button"
                htmlFor="menu-open"
              >
                <span
                  className="hamburger hamburger-1"
                />
                <span
                  className="hamburger hamburger-2"
                />
                <span
                  className="hamburger hamburger-3"
                />
              </label>,
              <div
                className="nav-list"
              >
                <Link
                  onClick={[Function]}
                  replace={false}
                  to="/buy"
                >
                  <i
                    className="fas fa-carrot"
                  >
                    <p>
                      buy
                    </p>
                  </i>
                </Link>
                <Link
                  replace={false}
                  to="/profile"
                >
                  <i
                    className="fas fa-user-circle"
                  >
                    <p>
                      my profile
                    </p>
                  </i>
                </Link>
                <Link
                  replace={false}
                  to="/add-product"
                >
                  <i
                    className="fas fa-plus-circle"
                  >
                    <p>
                      inventory
                    </p>
                  </i>
                </Link>
                <Link
                  onClick={[Function]}
                  replace={false}
                  to="/"
                >
                  <i
                    className="fas fa-sign-out-alt"
                  >
                    <p>
                      sign out
                    </p>
                  </i>
                </Link>
              </div>,
            ],
            "className": "menu",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "menu-open",
                "href": "#",
                "id": "menu-open",
                "name": "menu-open",
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="hamburger hamburger-1"
                  />,
                  <span
                    className="hamburger hamburger-2"
                  />,
                  <span
                    className="hamburger hamburger-3"
                  />,
                ],
                "className": "menu-open-button",
                "htmlFor": "menu-open",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "hamburger hamburger-1",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "hamburger hamburger-2",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "hamburger hamburger-3",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    onClick={[Function]}
                    replace={false}
                    to="/buy"
                  >
                    <i
                      className="fas fa-carrot"
                    >
                      <p>
                        buy
                      </p>
                    </i>
                  </Link>,
                  <Link
                    replace={false}
                    to="/profile"
                  >
                    <i
                      className="fas fa-user-circle"
                    >
                      <p>
                        my profile
                      </p>
                    </i>
                  </Link>,
                  <Link
                    replace={false}
                    to="/add-product"
                  >
                    <i
                      className="fas fa-plus-circle"
                    >
                      <p>
                        inventory
                      </p>
                    </i>
                  </Link>,
                  <Link
                    onClick={[Function]}
                    replace={false}
                    to="/"
                  >
                    <i
                      className="fas fa-sign-out-alt"
                    >
                      <p>
                        sign out
                      </p>
                    </i>
                  </Link>,
                ],
                "className": "nav-list",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <i
                      className="fas fa-carrot"
                    >
                      <p>
                        buy
                      </p>
                    </i>,
                    "onClick": [Function],
                    "replace": false,
                    "to": "/buy",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <p>
                        buy
                      </p>,
                      "className": "fas fa-carrot",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "buy",
                      },
                      "ref": null,
                      "rendered": "buy",
                      "type": "p",
                    },
                    "type": "i",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <i
                      className="fas fa-user-circle"
                    >
                      <p>
                        my profile
                      </p>
                    </i>,
                    "replace": false,
                    "to": "/profile",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <p>
                        my profile
                      </p>,
                      "className": "fas fa-user-circle",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "my profile",
                      },
                      "ref": null,
                      "rendered": "my profile",
                      "type": "p",
                    },
                    "type": "i",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <i
                      className="fas fa-plus-circle"
                    >
                      <p>
                        inventory
                      </p>
                    </i>,
                    "replace": false,
                    "to": "/add-product",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <p>
                        inventory
                      </p>,
                      "className": "fas fa-plus-circle",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "inventory",
                      },
                      "ref": null,
                      "rendered": "inventory",
                      "type": "p",
                    },
                    "type": "i",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <i
                      className="fas fa-sign-out-alt"
                    >
                      <p>
                        sign out
                      </p>
                    </i>,
                    "onClick": [Function],
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <p>
                        sign out
                      </p>,
                      "className": "fas fa-sign-out-alt",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "sign out",
                      },
                      "ref": null,
                      "rendered": "sign out",
                      "type": "p",
                    },
                    "type": "i",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "nav",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
